testsuite/MDAnalysisTests/core/test_groups.py

```diff
@@ -885,6 +885,10 @@ def test_segmentgroup_repr(self, u):
     def test_segmentgroup_str(self, u):
         sg = u.segments[:10]
         assert str(sg) == '<SegmentGroup [<Segment 4AKE>]>'
+    
+    def test_atomgroupresidue_dtype(self,u):
+        rg = list(u.atoms[100:130].residues)
```

Comment:
If we know that this can fail then we want to properly fail the test:
```suggestion
       try:
           rg = list(u.atoms[100:130].residues)
       except ValueError:
           raise AssertionError("Issue #2687 failure")
```

Reply:
> we want to properly fail the test

Hmm, maybe we have different views on this--I don't really mind, but I rarely see custom handling instead of letting `pytest` report the exception? Maybe this is becoming more common in MDA?

Reply:
I can see an advantage of just letting the exception trigger the failure because in this way you get more information and we don't accidentally mask some other failure we didn't think about.

You also have plenty of experience with numpy/scipy so I am happy to defer to your opinion here.

@darmis007 – let's ignore my suggestion and go back to simply
```python
# If this fails with ValueError see gh-2687
rg = list(u.atoms[100:130].residues)
assert ...
```
(no special exception handling)